### PR TITLE
PLAT-3565 added local dev docker config

### PR DESCRIPTION
**Pull request by kiyanwang**
### this relates to talis/platform#3565

- [x] Add a Dockerfile that will build a local image for development
- [x] Add a Docker Compose file to launch a redis service and run the local development image
- [x] Update Unit / Integration tests with hardcoded redis connection details to optionally override with env vars.
- [x] add wiki page: https://github.com/talis/talis-node/wiki/Local-Development

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
+FROM talis/ubuntu:1404-latest
```
Comment (malcyL) [inline Dockerfile:1]: 14.04 is unsupported now - would 18.04 not be better at this point?

Comment (malcyL) [reply Dockerfile:1]: I do prefer ubuntu over alpine for this development containers though. :-)

Comment (kiyanwang) [reply Dockerfile:1]: I want to keep the dev environment the same as what we've deployed for echo since im debugging an issue on there.

**Review by malcyL — APPROVED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        python \
+        build-essential \
+        envconsul \
+    && rm -rf /var/lib/apt/lists/* \
+    && apt-get clean
+
+
+# Tidy up
+RUN apt-get -y autoremove && apt-get clean && apt-get autoclean && \
+    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
```
Comment (mrliptontea) [inline Dockerfile:16]: This should run together we the above RUN. 

Creating a layer that removes things only hides those files for the next layer, but they are still present in the Docker image (or at least unless the image is [`--squash`ed](https://docs.docker.com/engine/reference/commandline/image_build/)).

Comment (kiyanwang) [reply Dockerfile:16]: I'll change that, in fairness, I was copying what we have in many of our other Dockerfiles. We might need to retrospectively correct those as well.

**Review by mrliptontea — APPROVED** [review]

**Pull request merged**